### PR TITLE
Guido/ab8705 convert crowdmonitor to schema based table creation

### PR DESCRIPTION
**Pull request by gkoller**
Schema based table creation, proper SQL statement generation (no string formatting), using Operators in favor of literal SQL (eg rename table)

  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Use pathless program name in call to subprocess.run.
  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Work around PostgresOperator incorrect type annotation.
  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Better and safer SQL INSERT for crowdmonitor.
  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Simplify crowdmonitor SQL query.
  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Better type annotations for crowdmonitor.
  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Schema based table creating for crowdmonitor.
  * [AB#8705](https://dev.azure.com/CloudCompetenceCenter/089b38ee-0066-4b66-a36c-20744a0e4348/_workitems/edit/8705) Uppercase constants per PEP8.

### CODE REVIEW

File: src/dags/crowdmonitor.py [hunk 1]
```diff
@@ -86,81 +53,143 @@ def copy_data_from_dbwaarnemingen_to_masterdb(*args, **kwargs):
...
+    return len(rows)
+
+
+def rm_tmp_tables(
+    task_id_postfix: str, tables: Iterable[str] = (f"{TABLE_ID}{TMP_TABLE_POSTFIX}",)
+) -> PostgresOperator:
+    """Remove tmp tables."""
+    return PostgresOperator(
+        task_id=f"rm_tmp_tables{task_id_postfix}",
+        # Airflow is bizar in some ways. Its `PostgresOperator` specifies the `sql` argument to
```
Comment (lbam) [inline src/dags/crowdmonitor.py:127]:
```suggestion
        # Airflow is bizarre in some ways. Its `PostgresOperator` specifies the `sql` argument to
```

File: src/dags/crowdmonitor.py [hunk 2]
```diff
@@ -152,15 +149,15 @@ def copy_data_in_batch(fetch_iterator, periode="uur"):
...
     default_args=args,
     description="Crowd Monitor",
-    on_failure_callback=get_contact_point_on_failure_callback(dataset_id=dag_id),
+    on_failure_callback=get_contact_point_on_failure_callback(dataset_id=DAG_ID),
 ) as dag:
     create_temp_tables = PostgresOperator(
         task_id="create_temp_tables",
         sql=SQL_CREATE_TEMP_TABLE,
-        params=dict(base_table=table_id),
+        params={'base_table': TABLE_ID},
```
Comment (jjmurre) [inline src/dags/crowdmonitor.py]: Strange, these are single quotes? Default behaviour of black is to use double quotes, right?

Comment (gkoller) [reply src/dags/crowdmonitor.py]: That's strange indeed. Black, as run from the pre-commit hook, should have picked that up. Will fix.

File: src/dags/crowdmonitor.py (hunk 1 above)
Comment (gkoller) [reply src/dags/crowdmonitor.py:127]: Will fix

File: src/dags/crowdmonitor.py [hunk 3]
```diff
@@ -86,81 +53,143 @@ def copy_data_from_dbwaarnemingen_to_masterdb(*args, **kwargs):
...
         )
 
         masterdb_hook = PostgresHook()
         try:
-            masterdb_hook.run(insert_sql)
+            with closing(masterdb_hook.get_conn()) as conn:
+                with closing(conn.cursor()) as cur:
+                    logger.info("Executing SQL statement %s", insert_stmt.as_string(conn))
+                    execute_values(cur, insert_stmt, rows, page_size=batch_size)
+                conn.commit()
```
Comment (lbam) [inline src/dags/crowdmonitor.py:112]: I think this should work too:

```suggestion
            with closing(masterdb_hook.get_conn()) as conn, closing(conn.cursor()) as cur:
                logger.info("Executing SQL statement %s", insert_stmt.as_string(conn))
                execute_values(cur, insert_stmt, rows, page_size=batch_size)
                conn.commit()
```

File: src/dags/crowdmonitor.py (hunk 2 above)
Comment (gkoller) [reply src/dags/crowdmonitor.py]: It's even stranger. Can't find it in my version of `crowdmonitor`. It's old code.

Comment (gkoller) [reply src/dags/crowdmonitor.py]: Ah, now I get it. You are reviewing things per commit (as I recommended). This must have been the result of not wanting to do too many cleanups in one and the same commit. I probably did get many `flake8` and `mypy` errors and decided to address those in other commits. Hence I commited this with `--no-verify`, which also bypassed the `black` run. Later commits resolved this. Sorry for the confusion.

Anyway it has been addressed.

File: src/dags/crowdmonitor.py (hunk 3 above)
Comment (gkoller) [reply src/dags/crowdmonitor.py:112]: Definitely and is much cleaner. Will fix. Thanks.

File: src/dags/crowdmonitor.py [hunk 4]
```diff
@@ -61,64 +67,54 @@ def copy_data_from_dbwaarnemingen_to_masterdb() -> None:
...
-        insert_sql = f"""
-            INSERT INTO {TABLE_ID}{TMP_TABLE_POSTFIX}
-            (sensor, naam_locatie, periode, datum_uur, aantal_passanten, gebied, geometrie)
-                VALUES {items_sql};
-        """
 
         masterdb_hook = PostgresHook()
         try:
-            masterdb_hook.run(insert_sql)
+            with closing(masterdb_hook.get_conn()) as conn, closing(conn.cursor()) as cur:
```
Comment (jjmurre) [inline src/dags/crowdmonitor.py:108]: Nice, didn't know you could do that in one with statement.

**Review by jjmurre — APPROVED** [review]
LGTM

**Pull request merged**